cmd/consumer/main.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"encoding/base64"
+	"fmt"
+	"log"
+	"math/big"
+	"os"
+	"time"
+
+	"github.com/Clever/amazon-kinesis-client-go/kcl"
+
+	"github.com/Clever/kinesis-to-firehose/firehose"
+)
+
+type RecordProcessor struct {
+	shardID           string
+	sleepDuration     time.Duration
+	checkpointRetries int
+	checkpointFreq    time.Duration
+	largestSeq        *big.Int
+	largestSubSeq     int
+	lastCheckpoint    time.Time
+	firehoseWriter    *firehose.FirehoseWriter
+}
+
+func New() *RecordProcessor {
+	return &RecordProcessor{
+		sleepDuration:     5 * time.Second,
+		checkpointRetries: 5,
+		checkpointFreq:    60 * time.Second,
+	}
+}
+
+func (rp *RecordProcessor) Initialize(shardID string) error {
+	rp.shardID = shardID
+	rp.lastCheckpoint = time.Now()
```

Comment:
This follows the approach used in `awslabs` example record processors https://github.com/awslabs/amazon-kinesis-client-python/blob/master/samples/sample_kclpy_app.py#L50

We're just using it so that we don't attempt to Checkpoint too frequently.

```go
if time.Now().Sub(rp.lastCheckpoint) > rp.checkpointFreq {
   // ... checkpoint
}
```